google_guest_agent/events/metadata/metadata.go

```diff
@@ -52,11 +64,42 @@ func (mp *Watcher) ID() string {
 
 // Events returns an slice with all implemented events.
 func (mp *Watcher) Events() []string {
-	return []string{LongpollEvent}
+	return []string{ReadyEvent, LongpollEvent}
 }
 
-// Run listens to metadata changes and report back the event.
-func (mp *Watcher) Run(ctx context.Context, evType string) (bool, interface{}, error) {
+func (mp *Watcher) runReadyWatcher(ctx context.Context, evType string) (bool, interface{}, error) {
+	descriptor, err := mp.client.Get(ctx)
+	if err != nil {
+		return false, nil, fmt.Errorf("Failed to get metadata descriptor: %+v", err)
```

Comment:
Should we have similar logging logic as with `runLongpollWatcher` here? My only concern is `failedPrevious` is only being used one of the run functions instead of both.

This isn't a requirement, but maybe error logging and handling should happen outside of both `runReadyWatcher` and `runLongpollWatcher`.

Reply:
It's a one shot only, we are not retrying (metadata package does) so we'll not exactly running the risk of having duplicated log messages. If we decide to implement a retry in this level then sure we should.